yspryn/hw6/min_max.py

```diff
@@ -0,0 +1,94 @@
+def min_ysp(*args, key=None):
+    res = args[0]
+    if key:
+        for i in args:
+            if key(i) < key(res):
+                res = i
+    else:
+        for i in args:
+            if i < res:
+                res = i
+    return res
+
+
+def max_ysp(*args, key=None):
+    res = args[0]
+    if key:
+        for i in args:
+            if key(i) > abs(res):
+                res = i
+    else:
+        for i in args:
+            if i > res:
+                res = i
+    return res
+
+
+def sorted_bubbles_method(*args, key=None, reverse=False):
+    list_to_sort = list(args)
+    stop_mark = True
+    if key:
+        while stop_mark:
+            stop_mark = False
+            for i in range(len(list_to_sort) - 1):
+                if reverse:
+                    temp_var = (list_to_sort[i] if key(list_to_sort[i]) < key(list_to_sort[i + 1]) else None)
+                else:
+                    temp_var = (list_to_sort[i] if key(list_to_sort[i]) > key(list_to_sort[i + 1]) else None)
+                if temp_var is not None:
+                    list_to_sort[i] = list_to_sort[i + 1]
+                    list_to_sort[i + 1] = temp_var
+                    stop_mark = True
+    else:
+        while stop_mark:
+            stop_mark = False
+            for i in range(len(list_to_sort) - 1):
+                if reverse:
+                    temp_var = (list_to_sort[i] if list_to_sort[i] < list_to_sort[i + 1] else None)
+                else:
+                    temp_var = (list_to_sort[i] if list_to_sort[i] > list_to_sort[i + 1] else None)
+                if temp_var is not None:
+                    list_to_sort[i] = list_to_sort[i + 1]
+                    list_to_sort[i + 1] = temp_var
+                    stop_mark = True
+    return list_to_sort
+
+
+def sorted_mim_max_method1(*args, key=None, reverse=False):
+    list_to_sort = list(args)
+    if reverse:
+        if key is None:
```

Comment:
no need to handle key is none or not none condition, 
since max and min functions can handle that for you

Reply:
do not duplicate code,
first select function - either it is min or it is max, and use it in your code

```
min_or_max = max if reverse else min

min_or_max([1, 2, 5, 3, 2, 1])


```